### PR TITLE
Add format rules for dune files

### PR DESCRIPTION
**Pull request by emillon**
This contains several related changes:

- `unstable-fmt` is renamed to `format-dune-file`
- remove `--inplace`
- some modules are renamed (`Dune_fmt` -> `Format_sexp`, `Fmt_cmd` -> `Format_dune_file`)
- a new version for `(using fmt)` is added: when 1.1 is used, dune files get auto-formatted.

### CODE REVIEW

**Comment by diml** [comment]
Seems good, but why remove `--inplace`?

File: src/format_sexp.mli [hunk 1]
```diff
@@ -0,0 +1,7 @@
+open Import
```
Comment (diml) [inline src/format_sexp.mli]: What do you think of calling this file `Format_dune_lang`?

**Comment by emillon** [comment]
`format-dune-file` is a plumbing command: it will be mostly used through `@fmt`. So I don't think it's useful to add an extra way to modify the files (it can be done using promotion).

File: src/format_sexp.mli (hunk 1 above)
Comment (emillon) [reply src/format_sexp.mli]: Sounds good to me.

**Comment by diml** [comment]
Ok

**Review by ghost — APPROVED** [review]

**Pull request merged**